hazelcast-client/src/main/java/com/hazelcast/client/cache/impl/NearCachedClientCacheProxy.java

```diff
@@ -205,11 +205,9 @@ protected void onPutIfAbsentAsyncInternal(K key, V value, Data keyData, Data val
     @Override
     protected <T> void onGetAndRemoveAsyncInternal(K key, Data keyData, ClientDelegatingFuture<T> delegatingFuture,
                                                    ExecutionCallback<T> callback) {
-        try {
-            super.onGetAndRemoveAsyncInternal(key, keyData, delegatingFuture, callback);
-        } finally {
-            invalidateNearCache(serializeKeys ? keyData : key);
-        }
+        Object callbackKey = serializeKeys ? keyData : key;
+        InvalidateCallback<T> wrapped = new InvalidateCallback<T>(callbackKey, callback);
```

Comment:
Like in the other methods, we should use a proper callback here, to invalidate the entries when the future is finished, not directly after this method returns.